app/src/candid/nns_parser.c

```
@@ -366,6 +366,123 @@ __Z_INLINE parser_error_t readCommandRefreshVotingPower(__Z_UNUSED parser_contex
+    txn->element.variant_index = 1;
+    CHECK_PARSER_ERR(readCandidInnerElement(txn, &txn->element))
+    if (txn->element.field_hash != hash_field_disburse_account) {
+        return parser_unexpected_field;
+    }
+    CHECK_PARSER_ERR(getCandidTypeFromTable(txn, txn->element.implementation))
+    CHECK_PARSER_ERR(readCandidOptional(txn))
+    if (txn->element.implementation != Null) {
+        CHECK_PARSER_ERR(getCandidTypeFromTable(txn, txn->element.implementation))
+        if (txn->txn_type != Record) {
+            return parser_unexpected_type;
+        }
+    }
+
+    // Field 2: percentage_to_disburse (Nat32)
+    CHECK_PARSER_ERR(getCandidTypeFromTable(txn, disburseRoot))
+    CHECK_PARSER_ERR(readCandidRecordLength(txn))
+    txn->element.variant_index = 2;
+    CHECK_PARSER_ERR(readCandidInnerElement(txn, &txn->element))
+    if (txn->element.field_hash != hash_percentage_to_disburse) {
+        return parser_unexpected_field;
+    }
+    if (txn->element.implementation != Nat32) {
+        return parser_unexpected_type;
+    }
+
+    // Now let's read
+    // Field 0 data: to_account_identifier (opt AccountIdentifier)
+    CHECK_PARSER_ERR(readCandidByte(ctx, &val->command.disburseMaturity.has_to_account_identifier))
+    if (val->command.disburseMaturity.has_to_account_identifier) {
+        CHECK_PARSER_ERR(readCandidText(ctx, &val->command.disburseMaturity.to_account_identifier))
+    }
+
+    // Field 1 data: to_account (opt Account)
+    uint8_t has_to_account_optional = 0;
+    CHECK_PARSER_ERR(readCandidByte(ctx, &has_to_account_optional))
+    if (has_to_account_optional) {
+        // Read Account.owner (optional Principal)
+        CHECK_PARSER_ERR(readCandidByte(ctx, &val->command.disburseMaturity.to_account.has_owner))
+        if (val->command.disburseMaturity.to_account.has_owner) {
+            uint8_t has_principal = 0;
+            CHECK_PARSER_ERR(readCandidByte(ctx, &has_principal))
+            if (has_principal) {
```

Is it possible that `has_owner` is true and `has_principal` is false? Should we return an error when that happens?